s3/pkg/service/object.go

```diff
@@ -982,9 +1127,42 @@ func (s *s3Service) ListObjects(ctx context.Context, in *pb.ListObjectsRequest,
 		}
 		object.StorageClass, _ = GetNameFromTier(obj.Tier, utils.OSTYPE_OPENSDS)
 		objects = append(objects, &object)
-		log.Debugf("object:%+v\n", object)
+		log.Infof("object:%+v\n", object)
+
+		objList, ok := mapNameToListOfObjects[object.ObjectKey]
+		if !ok {
+			// add a new obj list
+			objList = pb.ListObjects{
+				Objects:              make([]*pb.Object, 0),
+				XXX_NoUnkeyedLiteral: struct{}{},
+				XXX_unrecognized:     nil,
+				XXX_sizecache:        0,
+			}
+			objList.Objects = append(objList.Objects, &object)
+			mapNameToListOfObjects[object.ObjectKey] = objList
+		} else {
+			objList.Objects = append(objList.Objects, &object)
+			mapNameToListOfObjects[object.ObjectKey] = objList
+		}
+	}
+
+	out.ListOfListOfObjects = make([]*pb.ListObjects, len(mapNameToListOfObjects))
```

Comment:
Does this be compatible with AWS S3, otherwise many s3 client tools cannot be used.